### PR TITLE
Static Notification Payloads, Move more to `IdentifiedArray`

### PR DESCRIPTION
**Pull request by LePips**
- Overhauls current notifications to allow static payloads, so no more casting the user info is necessary.
- Some refactoring in `SeriesItemViewModel` and `SeasonItemViewModel` to use `IdentifiedArray`. Instead of passing around objects we look at IDs.

### CODE REVIEW

File: Shared/ViewModels/ItemViewModel/ItemViewModel.swift [hunk 1]
```diff
@@ -90,22 +90,44 @@ class ItemViewModel: ViewModel, Stateful {
...
-                    } else if let seriesID = userInfo["seriesID"], seriesID == item.id {
-                        Task { [weak self] in
-                            await self?.send(.backgroundRefresh)
-                        }
-                    }
-                } else if let newItem = notification.object as? BaseItemDto, newItem.id == self?.item.id {
-                    Task { [weak self] in
-                        await self?.send(.replace(newItem))
-                    }
+//        Notifications[.itemMetadataDidChange].publisher
```
Comment (JPKribs) [inline Shared/ViewModels/ItemViewModel/ItemViewModel.swift:93]: Do we need this commented code or can we delete this?

File: Swiftfin tvOS/Views/SelectUserView/SelectUserView.swift [hunk 2]
```diff
@@ -310,33 +310,27 @@ struct SelectUserView: View {
...
+                } else {
+                    serverSelection = .all
                 }
-
-                // change splash screen selection if necessary
-//                selectUserAllServersSplashscreen = serverSelection
             }
+
+            // change splash screen selection if necessary
+//            selectUserAllServersSplashscreen = serverSelection
```
Comment (JPKribs) [inline Swiftfin tvOS/Views/SelectUserView/SelectUserView.swift:333]: Same as above.

**Review by JPKribs — DISMISSED** [review]
@LePips Looks good minus the 2 comments I tagged. If we don't think we're going to use this later, I say we remove it and we can revisit version history if we ever need to reference it.

You know better than I do if it makes sense to keep those as comments. I pulled this and I didn't notice anything functionally different which is great!

File: Shared/ViewModels/ItemViewModel/ItemViewModel.swift (hunk 1 above)
Comment (LePips) [reply Shared/ViewModels/ItemViewModel/ItemViewModel.swift]: Good catch on this one, forgot to clean up.

File: Swiftfin tvOS/Views/SelectUserView/SelectUserView.swift (hunk 2 above)
Comment (LePips) [reply Swiftfin tvOS/Views/SelectUserView/SelectUserView.swift:333]: This was previously commented because this isn't complete on tvOS.

**Review by JPKribs — APPROVED** [review]

**Pull request merged**